plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -556,31 +580,162 @@ public void dropView(ConnectorSession session, SchemaTableName schemaViewName)
+        }
     }
 
     @Override
-    public void createMaterializedView(ConnectorSession session, SchemaTableName schemaViewName, ConnectorMaterializedViewDefinition definition, boolean replace, boolean ignoreExisting)
+    public void createMaterializedView(
+            ConnectorSession session,
+            SchemaTableName viewName,
+            ConnectorMaterializedViewDefinition definition,
+            boolean replace,
+            boolean ignoreExisting)
     {
-        throw new TrinoException(NOT_SUPPORTED, "createMaterializedView is not supported for Iceberg Glue catalogs");
+        Optional<com.amazonaws.services.glue.model.Table> existing = getTable(viewName);
+
+        // It's a create command where the materialized view already exists and 'if not exists' clause is not specified
+        if (!replace && existing.isPresent()) {
+            if (ignoreExisting) {
+                return;
+            }
+            throw new TrinoException(ALREADY_EXISTS, "Materialized view already exists: " + viewName);
+        }
+
+        existing.ifPresent(table -> dropMaterializedView(session, table));
+
+        // Create the storage table
+        SchemaTableName storageTable = createMaterializedViewStorageTable(session, viewName, definition);
+        // Create a view indicating the storage table
+        TableInput materializedViewTableInput = getMaterializedViewTableInput(
+                viewName.getTableName(),
+                encodeMaterializedViewData(fromConnectorMaterializedViewDefinition(definition)),
+                session.getUser(),
+                createMaterializedViewProperties(session, storageTable.getTableName()));
+        createTable(viewName.getSchemaName(), materializedViewTableInput);
     }
 
     @Override
-    public void dropMaterializedView(ConnectorSession session, SchemaTableName schemaViewName)
+    public void dropMaterializedView(ConnectorSession session, SchemaTableName viewName)
+    {
+        com.amazonaws.services.glue.model.Table view = getTable(viewName)
+                .orElseThrow(() -> new MaterializedViewNotFoundException(viewName));
+
+        dropMaterializedView(session, view);
+    }
+
+    private void dropMaterializedView(ConnectorSession session, com.amazonaws.services.glue.model.Table view)
     {
-        throw new TrinoException(NOT_SUPPORTED, "dropMaterializedView is not supported for Iceberg Glue catalogs");
+        String schemaName = view.getDatabaseName();
+        String viewName = view.getName();
+        String storageTableName = view.getParameters().get(STORAGE_TABLE);
+        if (storageTableName != null) {
+            try {
+                dropTable(session, new SchemaTableName(schemaName, storageTableName));
+            }
+            catch (TrinoException e) {
+                LOG.warn(e, "Failed to drop storage table '%s' for materialized view '%s'", storageTableName, viewName);
+            }
+        }
+        deleteTable(schemaName, viewName);
```

Comment:
Silly question, does it make sense to delete table that was not dropped correctly?